specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json

```diff
@@ -0,0 +1,2964 @@
+        }
+      },
+      "description": "Represents a resource name availability."
+    },
+    "Identity": {
+      "x-ms-client-flatten": true,
+      "type": "object",
+      "properties": {
+        "principalId": {
+          "type": "string",
+          "description": "ObjectId from the KeyVault",
+          "readOnly": true
+        },
+        "tenantId": {
+          "type": "string",
+          "description": "TenantId from the KeyVault",
+          "readOnly": true
+        },
+        "type": {
+          "type": "string",
+          "description": "Type of managed service identity.",
+          "enum": [
+            "UserAssigned"
+          ],
+          "x-ms-enum": {
+            "name": "ManagedServiceIdentityType",
+            "modelAsString": false
+          }
+        },
+        "userAssignedIdentities": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "object",
+            "description": "",
+            "items": {
+              "$ref": "#/definitions/UserAssignedIdentity"
+            }
+          },
+          "description": "Metadata of user assigned identity."
+        }
+      },
+      "description": "Properties to configure Identity for Bring your Own Keys"
+    },
+    "UserAssignedIdentity": {
+      "type": "object",
+      "properties": {
+        "principalId": {
+          "description": "Principal Id of user assigned identity",
+          "type": "string",
+          "x-ms-client-name": "PrincipalId",
+          "readOnly": true
+        },
+        "clientId": {
+          "description": "Client Id of user assigned identity",
+          "type": "string",
+          "x-ms-client-name": "ClientId",
+          "readOnly": true
+        }
+      },
+      "description": "Metadata of user assigned identity."
+    },
+    "CutoverMigrationRequest": {
+      "type": "object",
+      "description": "Cutover migration request",
+      "properties": {
+        "dnsChangeMode": {
+          "type": "string",
+          "enum": [
+            "Update",
+            "Revert",
+            "None"
+          ],
+          "x-ms-enum": {
+            "name": "DnsChangeMode",
+            "modelAsString": true
+          },
+          "description": "The DNS Mode"
+        }
+      }
+    },
+    "MigrationAssessmentRequest": {
+      "type": "object",
+      "properties": {
+        "sourceServerResourceId": {
+          "type": "string",
+          "description": "Source Server Resource Id to verify and perform migration assessment on."
+        }
+      },
+      "description": "Request from client to perform migration assessment"
+    },
+    "MigrationAssessmentResource": {
+      "type": "object",
+      "properties": {
+        "sourceServerResourceId": {
+          "type": "string",
+          "description": "Source Server Resource Id to verify and perform migration assessment on."
+        },
+        "targetPlatformProviderId": {
```

Comment:
>targetPlatformProviderId [](http://example.com/codeflow?start=9&length=24)

What is returned in this property? Does it have a fixed set of values?

Reply:
The provider id and servertype is returned -> "/Microsoft.DBforMySQL/flexibleServers"
It is expected that the customers send "/Microsoft.DBforMySQL/flexibleServers", but incase they do not, we throw an exception

Reply:
If it's a constant string, why doesn't your backend just fill it in? Why is it passed back and forth to/from the client as a separate property that never changes?

Reply:
I got that wrong. It is not sent by the customer in the request body. We send it as part of the response. It represents what is the target server platform the customer is looking to migrate to

Reply:
OK, but there's only one possibility, right? I still don't see why the server needs to tell the client that the migration will be to a flexible server, since that's what they already asked about. Do you anticipate there will be other options in the future? If so, I think this should be a string enum with one value, maybe something like migrateTo: { mySqlFlexibleServer }. As you add more kinds of results, the enum would get additional values. If I've misunderstood, maybe you can clarify what you expect the client to do with this value.